### ISSUE TITLE
Can't edit macros after initial insert

### ISSUE THREAD

**Issue by vomus** [mikitex70/redmine_drawio#135]
Hi!
I inserted a macro from visual_editor button, it it all right

{{drawio_attach(RnD_task_1.drawio,size=512,hilight=#0000ff)}} 

and I can edit the diagram. However when redmine rendered my wiki page it showed the diagram of a rather small size. So I went to the editing and inserted size=512. After saving I get a macros error

drawio_attach ("\xD0" from ASCII-8BIT to UTF-8)

Is this a bug?

**Comment by mikitex70** [comment]
I @vomus, I've done some testing and the problem seems to be due to a non-ASCII character in the attachment name.
It will take me a while to figure out how I will fix it.

**Comment by vomus** [comment]
Will do this evening

**Comment by vomus** [comment]
I have tested the dev branch, the bug disappeared! This is great. The one thing, however, I probably do not understand. I set the 'drawio' drawing type but changing a 'size' parameter in the macro does not make drawing any bigger or smaller . Should it do that or 'size' means something different?

When I choose a 'png' drawing type it works as  expect it to. But where does it keep an original drawing for further editing?

**Comment by mikitex70** [comment]
Hi @vomus, diagrams in the `xml` or `drawio` formats are rendered by the _drawio viewer library_, which ignores the size of the size of the container element.
Instead, diagrams in the `png` or `svg` format are rendered by the browser, which uses this size to scale the image.
I have added a `inizialzoom` parameter (and relativo input in the dialogs) so you can set the initial zoom factor of the diagram (default is `100%`), which can be changed by the user using the _zoom in_ and _zoom out_ buttons on the viewer toolbar (if enabled).
Try to see if this solution meets your needs.

**Comment by vomus** [comment]
Yeah... It works... It would be cool by the way to be able to save zoom level that is chosen at zoom control on the toolbar.

And finally, export to svg/png when saving a page to pdf/html. Now if drawing are in drawio format they can not be exported into printable page...

**Comment by mikitex70** [comment]
No, al this is not feasible: the diagram viewer is this, a viewer, and it is made by the Draw.io team.
Maybe i can find a way to read the actual zoom level in the diagram, but I can't pass it to the edit page as a zoom default do be saved. The diagram viewer runs in page view, the editing is done in another page and this is how Redmine works.

The export in PNG/SVG has similar limitations: the viewer can only show diagrams, the export is done by the Draw.io editor. Maybe I can find an API on the Draw.io server to request the export ... but I'm not sure.

Printing the whole page with the diagram is also a problem: the diagram is rendered on the browser, the PDF is generated by the server, which cannot execute the viewer library (it is a client library, not a server library). If printing the diagram with the page is important there are two solutions:
* use the PNG or SVG formats, which are handled by the Redmine PDF library
* print the whole page using the browser, which prints the page as-is (zoomed and navigated by the user) and can export it as a PDF document.
